src/internal/client.ts

```diff
@@ -1139,6 +1139,73 @@ export class TypedClient {
     await this.makeRequestAsyncOmit(requestOptions, '', [204])
   }
 
+  /**
+   * this call will aggregate the parts on the server into a single object.
+   */
+  async completeMultipartUpload(
+    bucketName: string,
+    objectName: string,
+    uploadId: string,
+    etags: {
+      part: number
+      etag?: string
+    }[],
+  ): Promise<{ etag: string; versionId: string | null }> {
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
+    }
+    if (!isValidObjectName(objectName)) {
+      throw new errors.InvalidObjectNameError(`Invalid object name: ${objectName}`)
+    }
+    if (!isString(uploadId)) {
+      throw new TypeError('uploadId should be of type "string"')
+    }
+    if (!isObject(etags)) {
+      throw new TypeError('etags should be of type "Array"')
+    }
```

Comment:
As we now have types, do we still need these checks?

Reply:
there are some developers who doesn't use typescript

Reply:
I'm okay to remove these checks either

Reply:
we can leave it for now..